packages/main/src/Slider.js

```diff
@@ -0,0 +1,457 @@
+		const handleStartDomRect = handleStart.getBoundingClientRect();
+		const handleEndDomRect = handleEnd.getBoundingClientRect();
+		const inHandleStartDom = clientX >= handleStartDomRect.left && clientX <= handleStartDomRect.right;
+		const inHandleEndDom = clientX >= handleEndDomRect.left && clientX <= handleEndDomRect.right;
+		
+		// If the two handles are overlapping return the second (end) one as in general the more common drag move is to the right
+		if (inHandleStartDom && inHandleEndDom) {
+			return handleEnd;
+		}
+
+		// If one of the handle is pressed return that one
+		if (inHandleStartDom) { 	
+			return handleStart;
+		}
+		
+		if (inHandleEndDom) {
+			return handleEnd;
+		}
+
+		// If no handle is returned yet that means the press ocurred outside of both of them,
+		// return that handle that is closer to the press point
+		if (value < this.value) {
+			return handleStart;
+		}
+
+		if (value > this.endValue) {
+			return handleEnd;
+		}
+	}
+
+	/**
+	 * Sets the slider value from an event
+	 */
+	_calculateValueFromInteraction(event) {
+		const pageX = this._getPageXValueFromEvent(event);
+		const min = this.min;
+		const max = this.max;
+		const step = this.step;
+		let value = this._computeValueFromPageX(pageX);
+
+		// "Stepihfy" the raw value - calculate a step value
+		if (this.step !== 0) {
+			const numSteps = Math.round(value / step);
+			value = numSteps * step;
+		}
+
+		// Normalize value
+		if (value < min) {
+			value = min;
+		} else if (value > max) {
+			value = max;
+		}
+
+		return value;
+	}
+
+	/**
+	 * Gets pageX value from event on user interaction with the Slider
+	 */
+	_getPageXValueFromEvent(event) {
+		if (event.targetTouches && event.targetTouches.length > 0) {
+			return event.targetTouches[0].pageX;
+		}
+		return event.pageX;
+	}
+
+	_setValue(value) {
+			this.value = value;
+			this.fireEvent("input");
+	}
+
+	/**
+	 * Computes the new value (in %) from the pageX position of the cursor
+	 */
+	// TODO RTL support?
+	_computeValueFromPageX(pageX) {
+		const max = this.max;
+		const min = this.min;
+
+		// Determine pageX position relative to the Slider DOM
+		const xPositionRelative = pageX - this.boundingDOMRect.left;
+		// Calculate the percentage complete (the "progress")
+		const percentageComplete = xPositionRelative / this.boundingDOMRect.width;
+		// Fit (map) the complete percentage between the min/max value range
+		return min + percentageComplete * (max - min);
+	}
+
+	// Update UI after user interaction
+	_updateUI(value) {
+		const max = this.max;
+		const min = this.min;
+		const percentageComplete = (value - min) / (max - min);
+		const sliderDomRect = this.boundingDOMRect;
+		const translatePx = percentageComplete * sliderDomRect.width;
+		const handleDom = this.handle;
+
+		// Update the position of the handle depending on the value
+		// Center the Slider Handle position under the cursor/pointer
+		handleDom.style.setProperty("left", `${translatePx}px`);
```

Comment:
You have to check how we add inline styles usually in one of the existing components

For example

```js
// Input.js
get styles() {
		return {
			suggestionPopoverHeader: {
				"display": this._listWidth === 0 ? "none" : "inline-block",
				"width": `${this._listWidth}px`,
				"padding": "0.5625rem 1rem",
			},
...

// InputPopover.hbs

<ui5-responsive-popover style="{{styles.suggestionsPopover}}"
```